src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -115,6 +115,225 @@ BitFlags<Feature> WiFiFeatures(WiFiDriver * driver)
     return features;
 }
 
+/// Performs an auto-release of the given item, generally an `Iterator` type
+/// like Wifi or Thread scan results.
+template <typename T>
+class AutoRelease
+{
+public:
+    AutoRelease(T * iterator) : mValue(iterator) {}
+    ~AutoRelease()
+    {
+        if (mValue != nullptr)
+        {
+            mValue->Release();
+        }
+    }
+
+private:
+    T * mValue;
+};
+
+/// Convenience macro to auto-create a variable for you to release the given name at
+/// the exit of the current scope.
+#define DEFER_AUTO_RELEASE(name) AutoRelease autoRelease##__COUNTER__(name)
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_STATION || CHIP_DEVICE_CONFIG_ENABLE_WIFI_AP || CHIP_DEVICE_CONFIG_ENABLE_THREAD
```

Comment:
Why would we include wifi code just because CHIP_DEVICE_CONFIG_ENABLE_THREAD?

Reply:
Updated ... I had copied from where it is used, not sure how I got the thread bit. I blame autocomplete in the IDE.